aten/src/ATen/native/MathBitsFallback.cpp

```diff
@@ -0,0 +1,119 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <torch/library.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/native/UnaryOps.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+
+struct MathOpFallback {
```

Comment:
I wonder if "MathOpFallback" is really the right way to talk about this. I think the more important property you need for the algorithm here to be correct is for the op that we're doing the view on to be "self inverse", since we use the self inverse property to mean that we only need a single boolean to represent the state (false -f-> true -f-> false).

Reply:
Yes. But I don't think this class works with non-self-inverse but invertible ops. In particular, you need both `math_op_` and `math_op_inverse_` in that case. Nor do I think it makes sense to prematurely generalize this class to support it.